pkg/platform/packet/packet.go

```diff
@@ -294,3 +338,122 @@ func (c *config) checkWorkerPoolNamesUnique() hcl.Diagnostics {
+// can't work reliably.
+// For more info, see comment when calling terraformCreateReservations().
+func checkEachReservation(reservationIDs map[string]string, resDefault, nodeRole, name string) hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	errorPrefix := "Worker pool"
+	if nodeRole == "controller" {
+		errorPrefix = "Cluster"
+	}
+
+	// The following (several) checks try to avoid this: having a worker
+	// pool that a node uses specific UUID as hardware reservation ID and
+	// another node in the same pool that uses "next-available".
+	// All different variations that in the end result in that are checked
+	// below, and the reason is simple: we can't guarantee for those cases
+	// that nodes can be created reliably. Creation granularity is per pool,
+	// so if one pool mixes both, we can't guarantee that another pool
+	// created later that needs specific UUIDs won't have them used by the
+	// instances using "next-available" in the previous worker pool created.
+	// This can be solved in two ways: adding more granularity, or forbidding
+	// those cases. We opt for the second option, for simplicity, given that
+	// in the rare case that the user needs to mix them, it can specify another
+	// identical worker pool with "next-available".
+
+	// Avoid cases that set (to non default values) reservation_ids and
+	// reservation_ids_default.
+	if len(reservationIDs) > 0 && resDefault != "" {
+		diagnostics = append(diagnostics, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  fmt.Sprintf("%v can't set both: reservation_ids and reservation_ids_default", errorPrefix),
+			Detail:   fmt.Sprintf("%v: %v sets both, instead add an entry in reservations_ids for each node", errorPrefix, name),
+		})
+	}
+
+	// Check reservation_ids map doesn't use "next-available" as a value.
+	for _, v := range reservationIDs {
+		if v != "next-available" {
+			continue
+		}
+
+		diagnostics = append(diagnostics, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  fmt.Sprintf("%v reservations_ids entries can't use \"next-available\"", errorPrefix),
+			Detail:   fmt.Sprintf("%v: %v uses it, use specific UUIDs or reservations_ids_default only", errorPrefix, name),
+		})
+	}
+
+	// Check format is:
+	// controller-<int> or worker-<int>
+	// If not, terraform code will silently ignore it. We don't want that.
+	resPrefix := "worker-"
+	if nodeRole == "controller" {
+		resPrefix = "controller-"
+	}
+
+	d := checkResFormat(reservationIDs, name, errorPrefix, resPrefix)
+	diagnostics = append(diagnostics, d...)
+
+	return diagnostics
+}
+
+// checkResFormat checks that format for every key in reservationIDs is:
+// <resPrefix>-<int>.
+func checkResFormat(reservationIDs map[string]string, name, errorPrefix, resPrefix string) hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	for key := range reservationIDs {
+		hclErr := &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Format used in reservation_ids invalid",
```

Comment:
This error message sounds weird to me. How about:
```suggestion
			Summary:  "Invalid reservation ID",
```

or something?

Reply:
You are right, now that I look at it after a few days seems weird. Your suggestion seems more clear, will change it. Thanks!

Reply:
Fixed, thanks!